packages/workspace/src/browser/workspace-frontend-contribution.ts

```diff
@@ -418,14 +420,32 @@ export class WorkspaceFrontendContribution implements CommandContribution, Keybi
         } while (selected && exist && !overwrite);
         if (selected) {
             try {
-                await this.commandRegistry.executeCommand(CommonCommands.SAVE.id);
-                await this.fileService.copy(uri, selected, { overwrite });
+                await this.copySave(uri, selected, overwrite);
             } catch (e) {
                 console.warn(e);
             }
         }
     }
 
+    private async copySave(source: URI, selected: URI, overwrite: boolean): Promise<void> {
+        const sourceWidget = this.applicationShell.currentWidget;
+        const sourceSaveable = Saveable.get(sourceWidget);
+        if (sourceWidget && sourceSaveable && sourceSaveable.createSnapshot && sourceSaveable.revert) {
```

Comment:
Would it be desirable to give the user some indication that a save-as operation had failed if their particular brand of `Saveable` doesn't have `createSnapshot` or `revert`? As a practical matter, that's very unlikely to occur, but it looks like this code would just fail silently in that case, and that might be confusing.

Reply:
Sure, makes sense. Do you have a preference how this indicator should look like? (Notification?)

Reply:
Notification sounds reasonable to me.

Reply:
Using the `NotificationManager` leads to tsc errors (as seen in 5937ccc). Any idea how I can resolve these?

Reply:
It looks like you didn't add an explicit dependency on the `messages` package in the `workspace` package. If you add `@theia/messages` to the `packages/workspace/package.json` and a `reference` to the `messages` package in `packages/workspace/compile.tsconfig.json`, you should be able to use notifications.

Reply:
(It looks like that won't create any circularity, but if you'd rather not bother with that, you could show a toast using the `MessageService`, which is in `core` and so already accessible in the `workspace` package.)

Reply:
@Darkgaja please use the `MessageService` to display messages to end-users (in this case a warning or error might be the best option).
- https://eclipse-theia.github.io/theia/docs/next/classes/core.messageservice-1.html